cap-async-std/src/fs/dir.rs

```diff
@@ -66,13 +72,55 @@ impl Dir {
         path: P,
         options: &OpenOptions,
     ) -> io::Result<File> {
-        self.sys.open_file_with(path.as_ref(), options)
+        let file = unsafe { self.as_sync_file() };
+        Self::_open_file_with(&file, path.as_ref(), options)
+    }
+
+    #[cfg(any(unix, target_os = "fuchsia"))]
+    fn _open_file_with(
+        file: &std::fs::File,
+        path: &Path,
+        options: &OpenOptions,
+    ) -> io::Result<File> {
+        open(file, path.as_ref(), options).map(|f| unsafe { File::from_raw_fd(f.into_raw_fd()) })
+    }
+
+    #[cfg(windows)]
+    fn _open_file_with(
+        file: &std::fs::File,
+        path: &Path,
+        options: &OpenOptions,
+    ) -> io::Result<File> {
+        open(file, path.as_ref(), options)
+            .map(|f| unsafe { File::from_raw_handle(f.into_raw_handle()) })
```

Comment:
Could we add a `File::from_sync_file` and factor out the fd/handle parts so that Unix and Windows can otherwise share an `_open_file_with`?

Reply:
Actually, it turns out there's a `From` conversion provided by `async_std`, so we can use `File::from_std` in combination with `From` :-)